### ISSUE TITLE
TranslatableFields does not allow displaying a label for the fields inside

### ISSUE THREAD

**Issue by clement-escolano** [marmelab/react-admin#8480]
**What you were expecting:**
I would like to add a label on `TextField` inside a `TranslatableFields` component.

**What happened instead:**
No label is displayed.

**Steps to reproduce:**
Follow the documentation and use this code for instance:
```tsx
<TranslatableFields locales={['en', 'fr']}>
    <TextField source="name" />
    <TextField source="description" />
</TranslatableFields>
```

See the result in the show view: translated name and description are displayed in the same square which makes it difficult to know which is which.

**Related code:**

For some reason, forking the repo on Sandbox or stackblitz does not work.

<details>
    <summary>Details on the error without changing any code</summary>
    <img width="1512" alt="image" src="https://user-images.githubusercontent.com/11637509/206303857-4e8dc957-b8f3-4944-8024-06e951a199a4.png">
</details>

The label not displayed can be seen in the default [codesandbox](https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple).

<img width="588" alt="image" src="https://user-images.githubusercontent.com/11637509/206304954-e66c5a25-8222-4e2f-96dc-4aa152093481.png">

I can force the display of a label by using `Labeled`:

```tsx
<TranslatableFields locales={['en', 'fr']}>
    <Labeled source="name">
        <TextField source="name" />
    </Labeled>
    <Labeled source="description">
        <TextField source="description" />
    </Labeled>
</TranslatableFields>
```

However, the value is now the whole field (`{"en": "English value", "fr": "French value"}`) instead of the translated value.

**Environment**

* React-admin version: 4.3.4
* Last version that did not exhibit the issue (if applicable): NA
* React version: 18.2.0
* Browser: NA
* Stack trace (in case of a JS error): NA

**Comment by fzaninotto** [comment]
Thanks for the report! You're right, this is a regression introduced in 4.0 when we removed the `addLabel` prop.

**Comment by fzaninotto** [comment]
Fixed by #9126
